controllers/workspace/condition.go

```diff
@@ -40,19 +40,43 @@ type workspaceConditions struct {
 }
 
 func (c *workspaceConditions) setConditionTrue(conditionType dw.DevWorkspaceConditionType, msg string) {
+	if c.conditions == nil {
+		c.conditions = map[dw.DevWorkspaceConditionType]dw.DevWorkspaceCondition{}
+	}
+
 	c.conditions[conditionType] = dw.DevWorkspaceCondition{
 		Status:  corev1.ConditionTrue,
 		Message: msg,
 	}
 }
 
 func (c *workspaceConditions) setConditionFalse(conditionType dw.DevWorkspaceConditionType, msg string) {
+	if c.conditions == nil {
+		c.conditions = map[dw.DevWorkspaceConditionType]dw.DevWorkspaceCondition{}
+	}
+
 	c.conditions[conditionType] = dw.DevWorkspaceCondition{
 		Status:  corev1.ConditionFalse,
 		Message: msg,
 	}
 }
 
+func (c *workspaceConditions) copyFailedStatus(dwStatus dw.DevWorkspaceStatus) {
+	if dwStatus.Phase == dw.DevWorkspaceStatusFailed {
```

Comment:
Should not this func client check if status is failed before calling it?

Reply:
it actually copies FailedStart condition but not failed status. It's a bit confusing.

Reply:
Yeah, sorry, it was originally called copyConditionFromDevWorkspace - but i changed it to be used specifically for the failed condition (instead of needing to pass in the condition).

Reply:
it still check
```go
if dwStatus.Phase == dw.DevWorkspaceStatusFailed {
```

at the same time as all clients (1) has the same check

Reply:
I would refactor it in the following way https://github.com/sleshchenko/devworkspace-operator/commit/4fc74e794649cbb5edb0a34d7738e692e005bf93

WDYT?

Reply:
Looks good to me! I've updated the PR to reflect this